samples/v2/collected_artifacts.py

```diff
@@ -0,0 +1,138 @@
+from typing import List
```

Comment:
This sample is complex enough that a comment that summarizes / justifies it might be worthwhile. How many components does it have total? This is definitely above and beyond but an ASCII diagram might be nice.